riff-raff/app/restrictions/RestrictionChecker.scala

```diff
@@ -0,0 +1,49 @@
+package restrictions
+
+import com.gu.googleauth.UserIdentity
+import deployment.{ContinuousDeploymentRequestSource, RequestSource, UserRequestSource}
+
+object RestrictionChecker {
+  /** Method that runs one function if the supplied config is editable by the supplied source and another if it is not
+    * @param currentConfig The config to check - note that when a config is being checked this must be the original
+    *                      config not the updated config
+    * @param identity The identity of the user trying to make the edit
+    * @param ifEditable The function whose result will be returned if we consider the config to be editable
+    * @param ifNotEditable The function whose result will be returned if the config cannot be edited
+    * @tparam T The function return type
+    * @return The result of either ifEditable or ifNotEditable
+    */
+  def editable[T](currentConfig: Option[RestrictionConfig], identity: UserIdentity)
```

Comment:
Does this really save typing compared to returning a Boolean and using an if statement? If statements also take two blocks - the one to run if it is true and the one to run if it is not.